clang/lib/Index/ClangIndexRecordWriter.cpp

```diff
@@ -76,7 +75,9 @@ bool ClangIndexRecordWriter::writeRecord(StringRef Filename,
                                          std::string &Error,
                                          std::string *OutRecordFile) {
 
-  auto RecordHash = Hasher.hashRecord(IdxRecord);
+  std::array<uint8_t, 8> RecordHashArr = index::hashRecord(IdxRecord, Ctx);
```

Comment:
Wouldn’t it be easier if `index::hashRecord` returned `uint64_t` instead of `std::array<uint8_t, 8>`?

Reply:
I don't have a strong opinion here. This will have to be done either way, it's just a matter of where. I actually implemented it that way first, but ended up changing it since `std::array` is what the `HashBuilder` API returns. Happy to change if you'd prefer it.